src/Workspaces/Core/Portable/CodeStyle/CodeStyleHelpers.cs

```diff
@@ -111,5 +113,70 @@ public static bool TryParseNotification(string value, out NotificationOption not
             notification = NotificationOption.Silent;
             return false;
         }
+
+        public static Option<T> CreateOption<T>(
+            OptionGroup group,
+            string feature,
+            string name,
+            T defaultValue,
+            ImmutableArray<IOption>.Builder optionsBuilder,
+            params OptionStorageLocation[] storageLocations)
+        {
+            var option = new Option<T>(feature, group, name, defaultValue, storageLocations);
+            optionsBuilder.Add(option);
+            return option;
+        }
+
+        private static readonly CodeStyleOption<UnusedValuePreference> s_preferNoneUnusedValuePreference =
+            new CodeStyleOption<UnusedValuePreference>(UnusedValuePreference.None, NotificationOption.Silent);
+
+        private static readonly BidirectionalMap<string, UnusedValuePreference> s_unusedExpressionAssignmentPreferenceMap =
+            new BidirectionalMap<string, UnusedValuePreference>(new[]
+            {
+                KeyValuePairUtil.Create("none", UnusedValuePreference.None),
```

Comment:
we don't generally have a 'none' option for our preferences.  INstead, there's always a preference, but there can be a stated severity of 'none' indicating that there is nothing in the user experience pushing you toward such a style.  we shoudl likely follow that approach here.

Reply:
Thanks, removed.

---
In reply to: [230606383](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606383) [](ancestors = 230606383)

Reply:
👍